benchmark/src/main/scala/mesosphere/marathon/state/RootGroupBenchmark.scala

```diff
@@ -26,21 +28,45 @@ class GroupBenchmark {
         Container.Docker(Nil, "alpine", List(Container.PortMapping(2015, Some(0), 10000, "tcp", Some("thing")))))
     )
 
-  @Param(value = Array("2", "10", "100", "1000"))
+  def makeAppRaml(pathId: PathId) = Raml.toRaml(makeApp(pathId))
+
+  //@Param(value = Array("2", "10", "100", "1000"))
+  @Param(value = Array("10"))
   var appsPerGroup: Int = _
   lazy val appIds = 0 until appsPerGroup
 
-  @Param(value = Array("5", "10", "20"))
+  //@Param(value = Array("5", "10", "20"))
+  @Param(value = Array("5"))
   var groupDepth: Int = _
   lazy val groupIds = 0 until groupDepth
 
+  @Param(value = Array("5"))
+  var groupsPerLevel: Int = _
+
   lazy val groupPaths: Vector[AbsolutePathId] = groupIds.foldLeft(Vector[AbsolutePathId]()) { (allPaths, nextChild) =>
     val nextChildPath = (allPaths.lastOption.getOrElse(PathId.root) / s"group-$nextChild").asAbsolutePath
     allPaths :+ nextChildPath
   }
 
   lazy val rootGroup: RootGroup = fillRootGroup()
 
+  lazy val groupRaml: raml.GroupUpdate = {
+    raml.GroupUpdate(id = Some("/"), groups = Some(buildChildGroups(0, PathId.root)))
+  }
+
+  def buildChildGroups(level: Int, parent: AbsolutePathId): Set[raml.GroupUpdate] = {
+    if (level == groupDepth) return Set.empty
+
+    (0 to groupsPerLevel).map { gid =>
+      val groupPath = (parent / s"group-$gid").asAbsolutePath
```

Comment:
.asAbsolutePath isn't required anymore, parent is an AbsolutePath already
```suggestion
      val groupPath = (parent / s"group-$gid")
```